File: public/teien/actors/ghost_actor.js
GhostActor = function(param){
    teien.Actor.call(this, param);
    this.type = param.type;    
};
GhostActor.prototype = Object.create(teien.Actor.prototype);

GhostActor.prototype.setup = function(physics){
    teien.Actor.prototype.setup.call(this, physics);
};

GhostActor.prototype.fromHash = function(param){
    teien.Actor.prototype.fromHash.call(this, param);
};

GhostActor.prototype.toHash = function(){
    hash = teien.Actor.prototype.toHash.call(this);
    hash.type = this.type;    
    return hash;
};


GhostActor.prototype.update = function(params){
    
};

File: lib/teien/actors/ghost_actor.rb
require_relative '../actor'

module Teien
  class GhostActor <  Teien::Actor
    attr_accessor :mass
    
    def initialize(name, ext_info = nil)
      super(name)
      @type = "Ghost"
    end
    
    def setup(physics)
    end
    
    def to_hash
      hash = super
      hash[:type] = @type
      hash
    end
    
    def from_hash(hash)
      super(hash)
    end
  end
end
